### PR TITLE
[TGUI] Noticeboard

### PR DESCRIPTION
**Pull request by AyIong**
## What Does This PR Do
Transfer Noticeboard from clear HTML to TGUI.
Added a recipe to craft Noticeboard - 5 wood.
Slightly corrected 1 note sprite.

### About TGUI
This is the first UI which uses RMB. It doesn't have the usual buttons so I think it's worth telling you how it works.
**LMB** - Take note
**RMB** - Open full note (Paper window)
**LMB with pen in hand** - Edit note contents

## Why It's Good For The Game
Pretty(?) UI to replace the old HTML one
Also more opportunities to RP using the Notice Board since it can be crafted now

Closes - https://github.com/orgs/ParadiseSS13/projects/4?pane=issue&itemId=38641466

## Images of changes
qXCHsDgcLE

<details>
<summary>Demonstration</summary>

https://github.com/ParadiseSS13/Paradise/assets/69762909/a4b1a944-6657-49f8-a695-eda1cb3ab2b0

</details>

## Testing
Yes.
Found only 1 issue - UI doesn't refresh after editing a note on noticeboard.
I have no idea how to make it do that, because the paper code is completely confusing to me.

## Changelog
:cl:
add: You can craft Notice Board now. It costs 5 wood planks.
tweak: Notice Board is now uses TGUI.
/:cl:

### CODE REVIEW

**Comment by Oyu07** [comment]
https://static.wikia.nocookie.net/meme/images/a/a2/Maxresdefault_%283%29.jpg

File: code/game/objects/structures/noticeboard.dm [hunk 1]
```diff
@@ -1,82 +1,130 @@
...
-			icon_state = "nboard0[notices]"	//update sprite
-			to_chat(user, "<span class='notice'>You pin the paper to the noticeboard.</span>")
-		else
+	update_icon(UPDATE_ICON_STATE)
+
+/obj/structure/noticeboard/update_icon_state()
+	if(notices > 0)
+		icon_state = "noticeboard-[notices]"
+	else
+		icon_state = "noticeboard"
```
[the remark this answers is not in the document]
Comment (AyIong) [reply code/game/objects/structures/noticeboard.dm:45]: I think `if else` looks more readable

[the remark this answers is not in the document]
Comment (Oyu07) [reply code/game/objects/structures/noticeboard.dm:45]: Atleast do it 
`if(notices)` instead of `if(notices > 0)`

**Comment by AyIong** [comment]
> This pull request seems to be stale as there have been no changes in 14 days, please make changes within 7 days or the PR will be closed. If you believe this is a mistake, please inform a development team member on Discord.

bot

**Review by Oyu07 — APPROVED** [review]

**Review by Burzah — APPROVED** [review]

File: code/game/objects/structures/noticeboard.dm [hunk 2]
```diff
@@ -1,82 +1,143 @@
...
 	max_integrity = 150
 	var/notices = 0
 
+/obj/structure/noticeboard/New(turf/loc, direction, building = FALSE)
+	. = ..()
+	if(building)
+		setDir(direction)
+		set_pixel_offsets_from_dir(-32, 32, -30, 30)
+		update_icon(UPDATE_ICON_STATE)
+
```
Comment (AffectedArc07) [inline code/game/objects/structures/noticeboard.dm:30]: Curses to you

Comment (AyIong) [reply code/game/objects/structures/noticeboard.dm:30]: > Curses to you

image

**Pull request merged**